edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Diagnostics/Metric.cs

```diff
@@ -3,16 +3,20 @@
 namespace Microsoft.Azure.Devices.Edge.Agent.Diagnostics
 {
     using System;
+    using System.Collections.Generic;
+    using System.Linq;
     using Microsoft.Azure.Devices.Edge.Util;
 
     public sealed class Metric : IEquatable<Metric>
     {
         public DateTime TimeGeneratedUtc { get; }
         public string Name { get; }
         public double Value { get; }
-        public string Tags { get; }
+        public IReadOnlyDictionary<string, string> Tags { get; }
 
-        public Metric(DateTime timeGeneratedUtc, string name, double value, string tags)
+        int? metricKey;
```

Comment:
Why not init this in the ctor and not make it a nullable value at all?

Reply:
It's not used always, so I want it to be lazy. In this case, scraping never needs this value.

Reply:
may be fine, consider simplicity and readability too.

Reply:
Made it a Lazy<T>